app/models/participant.rb

```diff
@@ -118,6 +118,16 @@ def authorization
     authorization
   end
 
+  # Check if teammate is allowed to review today.
+  def teammate_review_allowed
+    # time when teammate review is allowed
+    due_date = assignment.current_due_date
+    assignment.find_current_stage == 'Finished' ||
+      due_date &&
+        (due_date.teammate_review_allowed_id == 3 ||
```

Comment:
Please assure that teammate_review_allowed_id is defined for all due dates (esp. those with really old assignments).